pkg/signup/signup_service.go

```diff
@@ -0,0 +1,103 @@
+package signup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	crtapi "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+
+	errors2 "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+)
+
+type SignupServiceConfiguration interface {
+	GetNamespace() string
+}
+
+type SignupService interface {
+	CreateUserSignup(ctx context.Context, username, userID string) (*crtapi.UserSignup, error)
+}
+
+type SignupServiceImpl struct {
+	Namespace string
+	Client    *kubeclient.CRTV1Alpha1Client
+}
+
+func NewSignupService(cfg SignupServiceConfiguration) (SignupService, error) {
```

Comment:
could we add some documentation to our public apis?

Reply:
Done